From 72b4bd921f584f19ab861ba7671b7105c409c34c Mon Sep 17 00:00:00 2001
From: Ben Greear <greearb@candelatech.com>
Date: Mon, 24 May 2021 14:27:21 -0700
Subject: [PATCH 02/15] intel-oot:  Add tools to help convert oot driver.

Replace backports funk with something that can compile in-kernel.

Signed-off-by: Ben Greear <greearb@candelatech.com>
---
 .../tools/iwlwifi_oot_to_kernel.bash          | 34 +++++++++++++++++++
 .../wireless/intel-oot/tools/iwlwifi_sub.pl   |  9 +++++
 2 files changed, 43 insertions(+)
 create mode 100755 drivers/net/wireless/intel-oot/tools/iwlwifi_oot_to_kernel.bash
 create mode 100755 drivers/net/wireless/intel-oot/tools/iwlwifi_sub.pl

diff --git a/drivers/net/wireless/intel-oot/tools/iwlwifi_oot_to_kernel.bash b/drivers/net/wireless/intel-oot/tools/iwlwifi_oot_to_kernel.bash
new file mode 100755
index 0000000000000..d1d6343aabbd6
--- /dev/null
+++ b/drivers/net/wireless/intel-oot/tools/iwlwifi_oot_to_kernel.bash
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+_PWD=$(pwd)
+
+function do_dir {
+    for i in `ls`
+      do
+      if [ $i == "." ]
+          then
+          continue
+      fi
+      if [ $i == ".." ]
+          then
+          continue
+      fi
+      if [ $i == "tools" ]
+          then
+          continue
+      fi
+      echo "Checking file: $i"
+      if [ -d $i ]
+          then
+          echo "Changing to directory: $i"
+          (cd $i && do_dir)
+      else
+          echo "Substituting file: $i"
+          $_PWD/tools/iwlwifi_sub.pl $i > /tmp/foo.txt || exit 1
+          cp /tmp/foo.txt $i
+      fi
+    done
+}
+
+
+do_dir
diff --git a/drivers/net/wireless/intel-oot/tools/iwlwifi_sub.pl b/drivers/net/wireless/intel-oot/tools/iwlwifi_sub.pl
new file mode 100755
index 0000000000000..1c8fb4e0cb729
--- /dev/null
+++ b/drivers/net/wireless/intel-oot/tools/iwlwifi_sub.pl
@@ -0,0 +1,9 @@
+#!/usr/bin/perl
+
+while (<>) {
+  s/CPTCFG/CONFIG_OOT_/g;
+  # Back out some things we don't want changed.
+  s/CONFIG_OOT_NL80211_TESTMODE/CONFIG_NL80211_TESTMODE/g;
+  s/CONFIG_OOT_MAC80211_DEBUGFS/CONFIG_MAC80211_DEBUGFS/g;
+  print;
+}
--
2.26.3
